config/fields/mixins/pagepicker.php

```diff
@@ -3,23 +3,35 @@
 return [
     'methods' => [
         'pagepicker' => function (array $params = []) {
-            $query = $params['query'] ?? null;
-            $model = $this->model();
-            $site  = $this->kirby()->site();
+            $query    = $params['query'] ?? null;
+            $subpages = $params['subpages'] ?? true;
+            $model    = $this->model();
+            $site     = $this->kirby()->site();
 
-            if ($query) {
-                $pages = $model->query($query, 'Kirby\Cms\Pages');
-                $self  = null;
+            if (empty($query) === false) {
+                if ($subpages === true) {
+                    if ($parent = $site->find($params['parent'] ?? null)) {
+                        $pages = $parent->children();
+                    } else {
+                        $pages  = $model->query($query, 'Kirby\Cms\Pages');
+                        $parent = $pages->parent();
+                    }
+                } else {
+                    $pages = $model->query($query, 'Kirby\Cms\Pages');
+                }
             } else {
                 if (!$parent = $site->find($params['parent'] ?? null)) {
                     $parent = $site;
                 }
-
+                
                 $pages = $parent->children();
+            }
+            
+            if ($subpages === true) {
```

Comment:
This should then check for `$subpages === true && $parent !== null` to make sure that the following code never runs without a valid parent object.